### ISSUE TITLE
source-mssql: upgrade connector is failing 

### ISSUE THREAD

**Issue by sivankumar86** [airbytehq/airbyte#28069]
### Connector Name

source-mssql

### Connector Version

1.1.0

### What step the error happened?

Updating the connector

### Revelant information

I am working on updating connection version from 1.0.0 to 1.1.0 and it works fine in 1.0.0 but, it fails in latest version with cast exception on CDC incremental set up.

Caused by: java.lang.ClassCastException: class java.nio.HeapByteBuffer cannot be cast to class [B (java.nio.HeapByteBuffer and [B are in module java.base of loader 'bootstrap')
[d97c7aaa_ba56_4a4c_94c3_88921c0e031d_logs_20483_txt.log](https://github.com/airbytehq/airbyte/files/11996127/d97c7aaa_ba56_4a4c_94c3_88921c0e031d_logs_20483_txt.log)

### Relevant log output

```shell
Caused by: java.lang.ClassCastException: class java.nio.HeapByteBuffer cannot be cast to class [B (java.nio.HeapByteBuffer and [B are in module java.base of loader 'bootstrap')
	at io.debezium.util.ApproximateStructSizeCalculator.getValueSize(ApproximateStructSizeCalculator.java:65) ~[debezium-core-2.2.0.Final.jar:2.2.0.Final]
	at io.debezium.util.ApproximateStructSizeCalculator.getStructSize(ApproximateStructSizeCalculator.java:45) ~[debezium-core-2.2.0.Final.jar:2.2.0.Final]
	at io.debezium.util.ApproximateStructSizeCalculator.getValueSize(ApproximateStructSizeCalculator.java:68) ~[debezium-core-2.2.0.Final.jar:2.2.0.Final]
	at io.debezium.util.ApproximateStructSizeCalculator.getStructSize(ApproximateStructSizeCalculator.java:45) ~[debezium-core-2.2.0.Final.jar:2.2.0.Final]
	at io.debezium.util.ApproximateStructSizeCalculator.getApproximateRecordSize(ApproximateStructSizeCalculator.java:32) ~[debezium-core-2.2.0.Final.jar:2.2.0.Final]
	at io.debezium.pipeline.DataChangeEvent.objectSize(DataChangeEvent.java:31) ~[debezium-core-2.2.0.Final.jar:2.2.0.Final]
	at io.debezium.connector.base.ChangeEventQueue.doEnqueue(ChangeEventQueue.java:223) ~[debezium-core-2.2.0.Final.jar:2.2.0.Final]
```

I did further debug on this . I have found that "1.0.17" version has breaking change. it works before this version.

### Contribute

- [ ] Yes, I want to contribute

**Comment by sivankumar86** [comment]
@sergio-ropero Could you shed some light on this issue ? i am facing this issue after patching refactor change. I have reverted the changes however, i need to upgrade to include other changes. 

https://github.com/airbytehq/airbyte/pull/24582

**Comment by sivankumar86** [comment]
[d97c7aaa_ba56_4a4c_94c3_88921c0e031d_logs_23247_txt.log](https://github.com/airbytehq/airbyte/files/12285647/d97c7aaa_ba56_4a4c_94c3_88921c0e031d_logs_23247_txt.log)

**Comment by broom9** [comment]
I'm seeing a similar error using Postgres CDC source.

* Airbyte version: 0.50.29
* Postgres source extension version: 3.1.7

Key error messages:

```
Stack Trace: org.apache.kafka.connect.errors.ConnectException: Error while processing event at offset {transaction_id=null, lsn_proc=212451729808, messageType=INSERT, lsn_commit=212451723360, lsn=212451729808, txId=228836398, ts_usec=1695301365614594}
	at io.debezium.pipeline.EventDispatcher.dispatchDataChangeEvent(EventDispatcher.java:262)
	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.lambda$processMessages$0(PostgresStreamingChangeEventSource.java:279)
	at io.debezium.connector.postgresql.connection.pgoutput.PgOutputMessageDecoder.decodeInsert(PgOutputMessageDecoder.java:432)
	at io.debezium.connector.postgresql.connection.pgoutput.PgOutputMessageDecoder.processNotEmptyMessage(PgOutputMessageDecoder.java:208)
	at io.debezium.connector.postgresql.connection.AbstractMessageDecoder.processMessage(AbstractMessageDecoder.java:41)
	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.deserializeMessages(PostgresReplicationConnection.java:556)
	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.readPending(PostgresReplicationConnection.java:548)
	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.processMessages(PostgresStreamingChangeEventSource.java:216)
	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.execute(PostgresStreamingChangeEventSource.java:178)
	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.execute(PostgresStreamingChangeEventSource.java:41)
	at io.debezium.pipeline.ChangeEventSourceCoordinator.streamEvents(ChangeEventSourceCoordinator.java:174)
	at io.debezium.pipeline.ChangeEventSourceCoordinator.executeChangeEventSources(ChangeEventSourceCoordinator.java:141)
	at io.debezium.pipeline.ChangeEventSourceCoordinator.lambda$start$0(ChangeEventSourceCoordinator.java:109)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ClassCastException: class java.nio.HeapByteBuffer cannot be cast to class [B (java.nio.HeapByteBuffer and [B are in module java.base of loader 'bootstrap')
	at io.debezium.util.ApproximateStructSizeCalculator.getValueSize(ApproximateStructSizeCalculator.java:65)
	at io.debezium.util.ApproximateStructSizeCalculator.getStructSize(ApproximateStructSizeCalculator.java:45)
	at io.debezium.util.ApproximateStructSizeCalculator.getValueSize(ApproximateStructSizeCalculator.java:68)
	at io.debezium.util.ApproximateStructSizeCalculator.getStructSize(ApproximateStructSizeCalculator.java:45)
	at io.debezium.util.ApproximateStructSizeCalculator.getApproximateRecordSize(ApproximateStructSizeCalculator.java:32)
	at io.debezium.pipeline.DataChangeEvent.objectSize(DataChangeEvent.java:31)
	at io.debezium.connector.base.ChangeEventQueue.doEnqueue(ChangeEventQueue.java:223)
	at io.debezium.connector.base.ChangeEventQueue.enqueue(ChangeEventQueue.java:180)
	at io.debezium.pipeline.EventDispatcher$StreamingChangeRecordReceiver.changeRecord(EventDispatcher.java:421)
	at io.debezium.pipeline.EventDispatcher$2.changeRecord(EventDispatcher.java:245)
	at io.debezium.relational.RelationalChangeRecordEmitter.emitCreateRecord(RelationalChangeRecordEmitter.java:79)
	at io.debezium.relational.RelationalChangeRecordEmitter.emitChangeRecords(RelationalChangeRecordEmitter.java:47)
	at io.debezium.connector.postgresql.PostgresChangeRecordEmitter.emitChangeRecords(PostgresChangeRecordEmitter.java:94)
	at io.debezium.pip
[2d66be58_c105_4985_891e_9e3f9e48e1f3_logs_65_txt.txt](https://github.com/airbytehq/airbyte/files/12685820/2d66be58_c105_4985_891e_9e3f9e48e1f3_logs_65_txt.txt)
eline.EventDispatcher.dispatchDataChangeEvent(EventDispatcher.java:225)
	... 17 more
```

Full log files attached.

**Comment by tom-haines** [comment]
Thanks for tip about the 1.0.17 being the breaking change.  [PR#26473](https://github.com/airbytehq/airbyte/pull/26473/files) caused the issue by exposing a downstream bug in debezium in the queue size estimator.  It was fixed in [this commit](https://github.com/debezium/debezium/commit/016daf6fd8712217caf0c716de64a2b60b4d02df), and the fix is in debezium 2.4.0+.  Airybyte currently uses debezium 2.2.0 (you can fix by rebuilding airbyte with the newer dependency).

**Comment by sivankumar86** [comment]
I just removed below property and rebuild the connector. I am not facing this issue anymore if it helps.

` props.setProperty("max.queue.size.in.bytes", BYTE_VALUE_256_MB);`

**Comment by tom-haines** [comment]
I updated deps.toml to debezium 2.4.0.Final which also fixes it 
[ms-sql-debezium.patch](https://github.com/airbytehq/airbyte/files/12827863/ms-sql-debezium.patch).

**Comment by rodireich** [comment]
source-mssql 2.0.3 was released in #32024.
Integrated with debezium 2.4.0
